tests/media_filename_test.go

```diff
@@ -138,16 +160,17 @@ func downloadForFilename(t *testing.T, c *client.CSAPI, mxcUri string, diffName
 	if err != nil {
 		t.Fatalf("Got err when parsing content disposition: %s", err)
 	}
-
-	if mediaType != "attachment" {
-		t.Fatalf("Found unexpected mediatype %s, expected attachment", mediaType)
+	filename, hasFilename := params["filename"]
+	if mediaType == "attachment" {
+		if hasFilename {
+			return filename, true
+		} else {
+			t.Fatalf("Content Disposition did not have filename")
+			return "", true
```

Comment:
Isn't there no need to return here after `t.Fatalf`?

Reply:
Correct as it panics so control flow stops. It doesn't hurt from a readability PoV though.